gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -266,6 +269,31 @@ public int readContent(ByteBuffer dst) throws IOException {
       return totalBytesRead;
     }
 
+    private int readFromByteChannel(ByteBuffer buffer) throws IOException {
+      Stopwatch stopwatch = Stopwatch.createStarted();
+      int numBytesRead = byteChannel.read(buffer);
+      readAPITrace(stopwatch, numBytesRead);
+      return numBytesRead;
+    }
+
+    private void readAPITrace(Stopwatch stopwatch, int bytesRead) {
+      long elapsedTime = stopwatch.elapsed(TimeUnit.MILLISECONDS);
+      if (storageOptions.isTraceLogEnabled()) {
+        Map<String, Object> jsonMap = new HashMap<>();
+        jsonMap.put("method", "java-storage-read");
+        jsonMap.put("bytesRead", bytesRead);
+        jsonMap.put("gcs-path", resourceId);
+        jsonMap.put("durationMs", elapsedTime);
+        captureTraces(jsonMap);
+      }
+    }
+
+    private void captureTraces(Map<String, Object> apiTraces) {
+      if (storageOptions.isTraceLogEnabled()) {
+        EventLoggingHttpRequestInitializer.logDetails(apiTraces);
```

Comment:
With this we are adding logging responsibilites to EventLoggingHttpRequestInitializer. This is a code smell.